javascript/javascript1/week1/housey-pricey.js

```diff
@@ -0,0 +1,94 @@
+
+          console.log(`******3rd exersis House price****`);
+        //1st Way
+      /*   function volumeInMetersClac(deep, hight, wide) {
+            const volumeInMeters = deep * wide * hight;
+            return volumeInMeters
+        }
+  function housePriceCalc(volumeInMeters, gardenSizeInM2) {
+            const housePrice = volumeInMeters * 2.5 * 1000 + gardenSizeInM2 * 300;
+            return housePrice
+        }
+        //peter
+        const petervolumeInMeters = volumeInMetersClac(10, 8, 10)
+        const peterhousePriceCalc = housePriceCalc(petervolumeInMeters, 100)
+        console.log(petervolumeInMeters);
+        console.log(peterhousePriceCalc);
+        const peterpayment = 2500000
+        if (peterhousePriceCalc > peterpayment) {
+            console.log(`Peter paied too much  `);
+        }else 
+          console.log(`Peter paied too little  `);
+
+*/
+///////////////////////////////////////////////////////////////////
+          //2nd Way with object
+          const housePeter ={
+            name:"Peter",
+            deep:8,
+            wide:10,
+            hight:10,
+            gardenSizeInM2:100,
+            housecosts:2500000,
+   
+            
+            housePice:function(){
+                    let housevolumes = this.deep * this.wide * this.hight;
+                    const housePriceCalc = housevolumes * 2.5 * 1000 + this.gardenSizeInM2 * 300;
+                    return housePriceCalc
+                 }
+            //      housevolumes:function(deep,wide,hight){
+            //     let volume = deep * wide * hight;
+            //      return volume
+            //      },
+                
+            //  housePice:function(housevolumes,gardenSizeInM2){
+            //         const housePiceCalc = volumeInMeters * 2.5 * 1000 + gardenSizeInM2 * 300;
+            //         return housePiceCalc
+            //      }
+            
+
+          }
+          const houseJulia ={
+            name:"Julia",
+            deep:5,
+            wide:11,
+            hight:8,
+            gardenSizeInM2:70,
+            housecosts:1000000,
+            housePice:function(){
```

Comment:
that's probably out of scope of what we've discussed in "javascript week1" so
1. I like idea of adding functions into the object -- that's basically object oriented programming 
2. but now we have a problem: this function is copy pasted between 2 structures. Copy-paste is not good, but we can solve the problem in following way
```
function housePrice() {
    cons housevolumes = this.deep * this.wide * this.hight;
    const housePriceCalc = housevolumes * 2.5 * 1000 + this.gardenSizeInM2 * 300;
     return housePriceCalc
}

const housePeter = {
    // all the properties  for Peter
   housePrice: housePrice
}
const houseJulia = {
    // all the properties for Julia
   housePrice: housePrice
}